pipeline/service/_topic_modeling.py

```diff
@@ -45,65 +44,59 @@
 # python -m spacy download fr_core_news_sm
 # python -m spacy download de_core_news_sm
 
-class TopicModellingService:
+class TopicModelingService:
     default_model_path: str = './models'
 
-    custom_stopwords: set = {
-        " ", "\x96", "the", "to", "of", "20", "minuten",
-    }
-
-    stopwords_german_full_url: str = 'https://raw.githubusercontent.com/solariz/german_stopwords/refs/heads/master/german_stopwords_full.txt'
-    stopwords_french_full_url: str = 'https://raw.githubusercontent.com/stopwords-iso/stopwords-fr/refs/heads/master/stopwords-fr.txt'
+    custom_stopwords: set = {" ", "\x96", "the", "to", "of", "20", "minuten"}
 
     @staticmethod
     def preprocess(corpus, language='german'):
-        logger.info(f"Preprocessing {len(corpus)} texts for topic modelling.")
-
-        # Stop words
-        stopwords_url = TopicModellingService.stopwords_german_full_url if language == 'german' else TopicModellingService.stopwords_french_full_url
-        with urllib.request.urlopen(stopwords_url) as f:
-            stopwords_full = f.read().decode('utf-8')
-        stop_words = set(stopwords.words(language)) | set(stopwords_full) | TopicModellingService.custom_stopwords
+        logger.info(f"Preprocessing {len(corpus)} texts for topic modeling.")
 
+        stop_words = set(stopwords.words(language)) | TopicModelingService.custom_stopwords
         processed_texts = []
         nlp = nlp_de if language == 'german' else nlp_fr
 
         for doc in corpus:
             # Tokenize the document
             doc = nlp(str(doc).lower())  # Lowercase and tokenize
-            tokens = [token.text for token in doc if not token.is_stop and not token.is_punct]
+            tokens = [token.text for token in doc if not token.is_stop and not token.is_punct and (token.pos_ == "NOUN" or token.pos_ == "PROPN" or token.pos_ == "PRON")]  # only nouns
 
             # Lemmatize words and remove stopwords
             lemmatized_tokens = [lemmatizer.lemmatize(token) for token in tokens if token not in stop_words]
-
             processed_texts.append(lemmatized_tokens)
 
         return processed_texts
 
     @staticmethod
-    def fit_lda(
+    def fit_model(
             texts,
             language,
             num_topics=5,
-            dataset_passes=5
+            dataset_passes=5,
+            technique: str = 'lda',
     ):
-        processed_texts = TopicModellingService.preprocess(texts, language)
-
-        logger.info(f"Fitting LDA for {language}.")
+        processed_texts = TopicModelingService.preprocess(texts, language)
+        model = None
```

Comment:
We don't want to initialize the model as none because this will pass a silent error on to the next stage.  
instead assert that the technique is valid.

Reply:
This just asserts that technique is set.   
It needs to be validated to be one of a valid set  

```py
valid_models = ["lda", "lsa"]
assert technique in valid_models, f"Please provide a valid model from: {a}"
```

Reply:
Ok, thanks. Changed that